packages/dappmanager/src/daemons/telegram/telegramCommands.ts

```diff
@@ -0,0 +1,127 @@
+import TelegramBot from "node-telegram-bot-api";
+import * as db from "../../db";
+import { logs } from "../../logs";
+import { bold, buildTelegramMessage, url } from "./buildTelegramMessage";
+import { commandsList } from "./commandsList";
+
+/**
+ * Polls for commands and responds.
+ * Channel ID is not needed to response to messages
+ * @param telegramToken
+ */
+export async function telegramCommands(bot: TelegramBot): Promise<void> {
+  try {
+    bot.startPolling();
+    // POLLING ERRORS
+    // 1. EFATAL if error was fatal e.g. network error
+    // 2. EPARSE if response body could not be parsed
+    // 3. ETELEGRAM if error was returned from Telegram servers
+    // ETELEGRAM: 409 Conflict  =>  More than one bot instance polling
+    // ETELEGRAM: 404 Not Found => wrong token or not found
+    bot.on("polling_error", error => {
+      throw Error(`${error.name}: ${error.message}`);
+    });
+
+    // Listen for any messages. If channel ID does not exists, it saves the channel ID
+    bot.on("message", async msg => {
+      if (msg.text && commandsList.includes(msg.text) === false)
+        try {
+          const chatId = msg.chat.id.toString();
+          const channelExists = checkIfChannelIdExists(chatId);
+          if (channelExists === false) {
+            const message = buildTelegramMessage({
```

Comment:
I would use a different format since this is not a notification but a message from the bot itself

Reply:
I do not see it as a notification neither, that's why I use the success header (there is a notification header as well). I think is good for user experience to see a  green tick on the screen (success header) when the channel ID is saved.